string model_name
string[] param_names
float64[] model_params
int32 first_point
int32 last_point
